core/src/test/java/com/linecorp/armeria/internal/common/util/PortUtilTest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.common.util;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+
+import org.junit.jupiter.api.Test;
+
+class PortUtilTest {
+
+    @Test
+    void shouldAcquireUnusedPort() throws IOException {
+        final int port = PortUtil.unusedTcpPort();
+        try (ServerSocket serverSocket = new ServerSocket()) {
+            // Make sure the ServerSocket is successfully bound to the port.
+            serverSocket.bind(new InetSocketAddress("127.0.0.1", port));
+        }
+    }
```

Comment:
This can always fail. If really need to use this instead of binding at port 0, we should enforce retries.

Reply:
Agreed. Let me mark it as `@FlakyTest`.